library/src/main/java/org/apache/apex/malhar/lib/fs/redshift/RedshiftJdbcTransactionableOutputOperator.java

```
@@ -0,0 +1,283 @@
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.fs.redshift;
+
+import java.sql.BatchUpdateException;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.malhar.lib.fs.FSRecordCompactionOperator;
+
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.lib.db.jdbc.AbstractJdbcTransactionableOutputOperator;
+
+/**
+ *  A generic implementation of AbstractJdbcTransactionableOutputOperator for Redshift which takes FSRecordCompactionOperator.OutputMetaData.
+ *  Load the data into the specified redshift tables from data files. The files can be located either in S3 or an Amazon EMR.
+ *  Specify the bucketName property if the file is located in S3 or specify the emrClusterId if the file is location in EMR.
+ *
+ * @displayName Redshift Output Operator
+ * @category Output
+ * @tags database, jdbc, redshift
+ */
+public class RedshiftJdbcTransactionableOutputOperator extends AbstractJdbcTransactionableOutputOperator<FSRecordCompactionOperator.OutputMetaData>
+{
+  private static final Logger logger = LoggerFactory.getLogger(RedshiftJdbcTransactionableOutputOperator.class);
+  @NotNull
+  private String tableName;
+  @NotNull
+  private String accessKey;
+  @NotNull
+  private String secretKey;
+  @NotNull
+  private String redshiftDelimiter;
+  private String region;
+  private boolean readFromS3 = true;
+  private String emrClusterId;
+  private String bucketName;
+  protected transient Statement stmt;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+    try {
+      stmt = store.getConnection().createStatement();
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  protected String getUpdateCommand()
+  {
+    return null;
```

Would it be better if we throw Unsupported Implementation exception instead of returning null?